flink-core/src/main/java/org/apache/flink/api/java/typeutils/PojoTypeInfo.java

```diff
@@ -70,10 +77,41 @@
 	private static final Pattern PATTERN_NESTED_FIELDS = Pattern.compile(REGEX_NESTED_FIELDS);
 	private static final Pattern PATTERN_NESTED_FIELDS_WILDCARD = Pattern.compile(REGEX_NESTED_FIELDS_WILDCARD);
 
+	private static final Map<Class<?>, Class<? extends TypeSerializer>> customSerializers = new HashMap<>();
+	private static final Map<Tuple2<ArrayList<Integer>, Class>, Class<? extends TypeComparator>> customComparators =
+		new HashMap<>();
+
 	private final PojoField[] fields;
 	
 	private final int totalFields;
 
+	/**
+	 * Register a custom serializer for a type. The precedence of the serializers
+	 * is the following (highest to lowest): Kryo, Avro, Custom, Generated, Flink.
+	 * The chosen serializer will be the first one from the list that is turned on.
```

Comment:
I think the wording "turned on" is a bit confusing, because it means different things for the different elements of this list.
- I guess for Kryo, Avro, and Generater, you mean `enableForceKryo`, `enableForceAvro`, `enableCodeGeneration` was called?
- For "Custom", you mean that this method was called for the particular type.
- For "Flink", you are talking about `PojoSerializer`? This is kind of always turned on. However, it is not always applicable (if the type is not a POJO), in which case we fall back to Kryo, even if it is not "turned on" in the above sense, which is also confusing.

Another problem is that I'm not sure what happens with Tuples (and basic types and other special types) if `enableForceKryo` was called. I guess they don't participate in this mess and always get serialized by their special built-in serializers?